mux.go

```diff
@@ -203,6 +209,10 @@ func (m *mux) Acquire() wire {
 }
 
 func (m *mux) Store(w wire) {
+	w.SetPubSubHooks(PubSubHooks{})
+	if w.Pipelining() {
+		w.DoMulti(context.Background(), cmds.UnsubscribeCmd, cmds.PUnsubscribeCmd, cmds.SUnsubscribeCmd)
```

Comment:
As far as I understand this is an attempt to clean up?

Should the error be handled here? Should this be issued with timeout? Worrying that upon error connection will be left subscribed to channels upon error.

Also, In this case Redis will send unsubscribe messages for all channels connection was subscribed to.

> When no channels are specified, the client is unsubscribed from all the previously subscribed channels. In this case, a message for every unsubscribed channel will be sent to the client.

I am not sure I'd like to get those in my app.

What I want on PUB/SUB finish is simply close the connection and do not return it to the pool. Re-using PUB/SUB connections does not make sense to me. Simply closing - resolves everything IMO.

Reply:
Yes, this is an attempt to clean up in case of putting the connection back into the pool.

The timeout and error are actually handled. The connection will actually be closed and removed after back to the pool if there is an error.

> What I want on PUB/SUB finish is simply close the connection and do not return it to the pool. Re-using PUB/SUB connections does not make sense to me. Simply closing - resolves everything IMO.

Sure, I can add `Close` to the `DedicatedClient` interface.